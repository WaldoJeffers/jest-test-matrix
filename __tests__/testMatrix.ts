import testMatrix from '../src/index'

describe('testMatrix', () => {
  it('should generate a pretty ASCII-table snapshot when toMatchSnapshot is called', () => {
    interface IOpts {
      col1: string;
      col2: number;
      col3: string;
    }

    const fn1 =  ({col1, col2, col3}: IOpts) => col1.charCodeAt(0) + col2.toString().charCodeAt(0) + col3.charCodeAt(0)

    const r = 
      testMatrix(
        fn1,
        {
          col1: ['a', 'b', 'c', 'd'],
          col2: [26, 8, 3],
          col3: ['Δ', 'ƒ']
        },
      )
    expect(r).toMatchSnapshot()
  })

})