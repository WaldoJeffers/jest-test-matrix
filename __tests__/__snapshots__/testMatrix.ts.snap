// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testMatrix should generate a pretty ASCII-table snapshot when toMatchSnapshot is called 1`] = `
.-----------------------------.
|             fn1             |
|-----------------------------|
| col3 | col2 | col1 | result |
|------|------|------|--------|
| Δ    |   26 | a    |   1063 |
|      |      | b    |   1064 |
|      |      | c    |   1065 |
|      |      | d    |   1066 |
|      |    8 | a    |   1069 |
|      |      | b    |   1070 |
|      |      | c    |   1071 |
|      |      | d    |   1072 |
|      |    3 | a    |   1064 |
|      |      | b    |   1065 |
|      |      | c    |   1066 |
|      |      | d    |   1067 |
| ƒ    |   26 | a    |    549 |
|      |      | b    |    550 |
|      |      | c    |    551 |
|      |      | d    |    552 |
|      |    8 | a    |    555 |
|      |      | b    |    556 |
|      |      | c    |    557 |
|      |      | d    |    558 |
|      |    3 | a    |    550 |
|      |      | b    |    551 |
|      |      | c    |    552 |
|      |      | d    |    553 |
'-----------------------------'
`;
